iconservice/iconscore/icon_score_api_generator.py

```diff
@@ -53,26 +58,22 @@ def generate(score_funcs: list) -> list:
         return api
 
     @staticmethod
-    def check_on_deploy(score_funcs: list) -> None:
-        for func in score_funcs:
-            if func.__name__ == ScoreApiGenerator.__API_TYPE_ON_INSTALL or \
-                    func.__name__ == ScoreApiGenerator.__API_TYPE_ON_UPDATE:
-                ScoreApiGenerator.__check_on_deploy_function(signature(func))
+    def check_on_deploy(context: 'IconScoreContext', score: 'IconScoreBase') -> None:
+        custom_funcs = [value for key, value in getmembers(score.__class__, predicate=isfunction)
+                        if key in ScoreApiGenerator.__on_deploy]
+        for func in custom_funcs:
+            ScoreApiGenerator.__check_on_deploy_function(context, signature(func))
 
     @staticmethod
-    def __check_on_deploy_function(sig_info: 'Signature') -> None:
+    def __check_on_deploy_function(context: 'IconScoreContext', sig_info: 'Signature') -> None:
         params = dict(sig_info.parameters)
         for param_name, param in params.items():
             if param_name == 'self' or param_name == 'cls':
                 continue
-            if param.kind != Parameter.VAR_KEYWORD:
+            if context.revision > REVISION_2:
```

Comment:
It should be update as follows to check the parameters validity of on_install/on_deploy at revision 2 or lower.
```suggestion
            if context.revision > REVISION_2 or param.kind != Parameter.VAR_KEYWORD:
```